doc/guides/dts/index.rst

```diff
@@ -346,44 +346,534 @@ Below is sample header content generated for this node, in
    #define DT_INST_0_NXP_KINETIS_SIM_LABEL             DT_NXP_KINETIS_SIM_40047000_LABEL
    #define DT_INST_0_NXP_KINETIS_SIM                   1
 
-Generated macro names follow the format ``DT_<node>_<property>``. For
-``DT_NXP_KINETIS_SIM_40047000_BASE_ADDRESS``, the node part is
-``NXP_KINETIS_SIM_40047000``, based on the compatible string that matched a
-binding for the node (``nxp,kinetis-sim``) and the node's unit address
-(``...@4004700``).
+All macros generated from devicetree start with ``DT_`` and use all-uppercase.
+Most macro names contain a node identifier, and macros that correspond to
+devicetree properties also contain an identifier for the property.
 
-The ``*_BASE_ADDRESS`` part of the identifier is a fixed identifier generated
-from the special ``reg`` property on the node. ``*_SIZE`` is also generated
-from ``reg``. Other suffixes, like ``*_LABEL``, are generated directly from the
-property name.
+Node identifiers in macro names
+-------------------------------
 
-The second macro (``DT_INST_0_NXP_KINETIS_SIM_BASE_ADDRESS``) is an alias for
-the first macro. The node identifier ``...INST_0_NXP_KINETIS_SIM_...`` means
-"the first node with compatible string ``nxp,kinetis-sim``.
+Macros generated from particular devicetree nodes start with ``DT_<node>``,
+where ``<node>`` identifies the node. Several identifiers are generated for
+nodes:
 
-Aliases are also generated from any properties in the ``/aliases`` node. Take
-the DTS fragment below as an example.
+``DT_(<bus>_)<compatible>_<unit-address>``
+    The compatible string for the node followed by its unit address. If the
+    node has several compatible strings, the first one that matched a binding
+    is used.
+
+    If the node appears on a bus (has ``on-bus:`` in its binding), then the
```

Comment:
Would be good to clearly state that if it doesn't appear on a bus, then this part is omitted. Also an example would be great.